api/v1alpha1/ionoscloudmachine_types.go

```diff
@@ -168,6 +168,18 @@ type IonosCloudMachineSpec struct {
 	//+optional
 	FailoverIP *string `json:"failoverIP,omitempty"`
 
+	// LanID is the ID of the LAN if an existing LAN is required.
+	// A typical use-case is when an existing LAN has a Failover Group and the FailoverIP configured.
+	// This field is used when the Failover Group is created in another LAN.
+	//
+	// If the machine is a control plane machine, this field will not be taken into account.
+	// +optional
+	// +immutable
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="lanID is immutable"
+	// +kubebuilder:validation:MinLength=1
+	//nolint:revive
+	LanID *string `json:"lanID,omitempty"`
```

Comment:
As the disabled `revive` linter correctly pointed out, Go conventions would require `LANID` here. It's also consistent with `errLANIDIsEmpty` used elsewhere.

However, looking at the `AdditionalNetworks []Network` field above, for consistency on the CRD level the field should actually be called `NetworkID`.

Reply:
I'm okay with NetworkID